flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/partitioner/ForwardPartitionerTest.java

```diff
@@ -29,31 +25,19 @@
 /**
  * Tests for {@link ForwardPartitioner}.
  */
-public class ForwardPartitionerTest {
-
-	private ForwardPartitioner<Tuple> forwardPartitioner;
-	private StreamRecord<Tuple> streamRecord = new StreamRecord<Tuple>(null);
-	private SerializationDelegate<StreamRecord<Tuple>> sd = new SerializationDelegate<StreamRecord<Tuple>>(
-			null);
+public class ForwardPartitionerTest extends StreamPartitionerTest {
 
 	@Before
 	public void setPartitioner() {
-		forwardPartitioner = new ForwardPartitioner<Tuple>();
-	}
-
-	@Test
-	public void testSelectChannelsLength() {
-		sd.setInstance(streamRecord);
-		assertEquals(1, forwardPartitioner.selectChannels(sd, 1).length);
-		assertEquals(1, forwardPartitioner.selectChannels(sd, 2).length);
-		assertEquals(1, forwardPartitioner.selectChannels(sd, 1024).length);
+		streamPartitioner = new ForwardPartitioner<>();
 	}
 
 	@Test
 	public void testSelectChannelsInterval() {
 		sd.setInstance(streamRecord);
-		assertEquals(0, forwardPartitioner.selectChannels(sd, 1)[0]);
-		assertEquals(0, forwardPartitioner.selectChannels(sd, 2)[0]);
-		assertEquals(0, forwardPartitioner.selectChannels(sd, 1024)[0]);
+
+		assertEquals(0, streamPartitioner.selectChannels(sd, 1)[0]);
```

Comment:
I'm not sure if it's worth, but you could define
```
assertSelectedChannel(int expectedChannel, StreamRecord record, int numberOfChannels) {
   serializationDelegate.setInstance(record);
   int[] actualResult = streamPartitioner.selectChannels(serializationDelegate, numberOfChannels);
   assertEquals(1, actualResult.length);
   assertEquals(expectedChannel, actualResult[0]);
}
```

Reply:
This define could not be reused directly by `ShufflePartitionerTest` and `RebalancePartitionerTest`. And it will not reduce the code lines totally for reuse ones. So I think it is not very worth doing. What do you think or you have other concerns of this define?